src/Voron/Data/Tables/TableValueCompressor.cs

```diff
@@ -288,85 +287,139 @@ private static void RecreateRecoveryDictionaries(IPagerLevelTransactionState obj
-                    using var zip = new ZipArchive(fs, ZipArchiveMode.Update);
-                    int lastWritten = 0;
-                    if (zip.Entries.Count > 0)
+                    Exception innerEx = default;
+                    try
+                    {
+                        try
+                        {
+                            using (var finalFileStream = File.Open(path, FileMode.OpenOrCreate, FileAccess.ReadWrite, FileShare.Read))
+                            using (var zip = new ZipArchive(finalFileStream, ZipArchiveMode.Update))
+                            {
+                                int lastWritten = 0;
+
+                                if (zip.Entries.Count > 0)
+                                    // Entries were added to the file from storage, which is in chronological order,
+                                    // ensuring that the last entry is the most recent.
+                                    lastWritten = int.Parse(Path.GetFileNameWithoutExtension(zip.Entries[^1].Name));
+
+                                Debug.Assert(lastWritten >= dictionaries.State.NumberOfEntries,
+                                    message: "The number of last written entry in recovery file must be equal to or greater than the total number of entries in the state. " +
+                                             "Any deviation from this is a bug.");
+
+                                if (lastWritten == dictionaries.State.NumberOfEntries)
+                                    continue;
+
+                                AppendNewDictionaryEntries(lastWritten, zip);
+                            }
+                        }
+                        catch (Exception ex)
+                        {
+                            innerEx = ex;
+
+                            if (Logger.IsInfoEnabled)
+                                Logger.Info(msg: $"An unexpected error occurred while attempting to read the archive '{path}'. " +
+                                                 $"The file will be recreated from scratch.", ex);
+
+                            File.Delete(path);
+                            using (var finalFileStream = File.Open(path, FileMode.CreateNew, FileAccess.ReadWrite, FileShare.Read))
+                            using (var zip = new ZipArchive(finalFileStream, ZipArchiveMode.Update))
+                            {
+                                AppendNewDictionaryEntries(lastWritten: 0, zip);
+                            }
+                        }
+                    }
+                    catch (Exception e)
                     {
-                        // we try, if not successful, we'll start from scratch
-                        int.TryParse(Path.GetFileNameWithoutExtension(zip.Entries[^1].Name), out lastWritten);
+                        var aggregateException = innerEx != null
+                            ? new AggregateException(e, innerEx)
+                            : new AggregateException(e);
```

Comment:
It is better not to aggregate if there is only one exception.

Reply:
Done